### PR TITLE
add a regression test RandomBlastAMR-GPU

### PR DESCRIPTION
**Pull request by chongchonghe**
### Description

Add two more regression tests: SedovAMR-GPU and RandomBlastAMR-GPU. Now, we have the following tests with the modules tested:

- Sedov-GPU: Hydro
- SedovAMR-GPU: Hydro + AMR
- ShockCloud-GPU: Hydro + Cooling
- RandomBlast-GPU: Hydro + Cooling + Gravity
- RandomBlastAMR-GPU: Hydro + Cooling + Gravity + AMR
- RadhydroShell-GPU: Hydro + Radiation
- DiskGalaxy-GPU (disabled until we fix the reproducibility issue): Hydro + Cooling + Gravity + AMR + Particles

### Related issues
None. 

### Checklist
_Before this pull request can be reviewed, all of these tasks should be completed. Denote completed tasks with an `x` inside the square brackets `[ ]` in the Markdown source below:_
- [ ] I have added a description (see above).
- [ ] I have added a link to any related issues (if applicable; see above).
- [ ] I have read the [Contributing Guide](https://github.com/quokka-astro/quokka/blob/development/CONTRIBUTING.md).
- [ ] I have added tests for any new physics that this PR adds to the code.
- [ ] *(For quokka-astro org members)* I have manually triggered the GPU tests with the magic comment `/azp run`.

### CODE REVIEW

**Review by BenWibking — DISMISSED** [review]
Looks fine. How long does it take to run?

**Comment by chongchonghe** [comment]
> Looks fine. How long does it take to run?

RandomBlastAMR-GPU only takes a few minutes to run. 

@BenWibking I also changed the boundary conditions for both RandomBlast-GPU and RandomBlastAMR-GPU. Please check if this is the correct thing to do.

**Comment by chongchonghe** [comment]
Full result: https://quokka-astro.github.io/regression-testing-results/2025-09-08-011/index.html

Hydro+Gravity is reproducible, Hydro+Gravity+AMR is not. 

I also removed DiskGalaxy test because it it not very helpful before we fix the reproducibility issue. It takes a very long time to run, and removing it makes manual regression test easier.

**Comment by BenWibking** [comment]
> @BenWibking I also changed the boundary conditions for both RandomBlast-GPU and RandomBlastAMR-GPU. Please check if this is the correct thing to do.

I don't see a change in the boundary conditions. I see you changed the Physics_Traits to turn self-gravity on, but I don't see any other changes.

**Comment by chongchonghe** [comment]
> > @BenWibking I also changed the boundary conditions for both RandomBlast-GPU and RandomBlastAMR-GPU. Please check if this is the correct thing to do.
> 
> I don't see a change in the boundary conditions. I see you changed the Physics_Traits to turn self-gravity on, but I don't see any other changes.

The changes are in the in files. I have to set use_periodic_bc=0 and is_periodic=0,0,0 , because it complained some B.C. is not compatible with gravity solver.

**Comment by chongchonghe** [comment]
This makes it reflective B.C. 

Anyway, this PR is ready for review and merge.

**Comment by BenWibking** [comment]
Ah ok. The gravity solver currently assumes that the boundary conditions are open (vacuum outside the box), so periodic doesn't work.

Reflecting boundaries will do something weird when the supernova explode near the boundary, but this is ok for a test problem.

A bigger issue is that the energy deposition when AMR is activated will be wrong if the explosion is near a coarse-fine boundary:
https://github.com/quokka-astro/quokka/blob/f0a7e9ec70685ef1abda7dd06ec8ba13d1dbe710/src/problems/RandomBlast/blast.cpp#L125

File: src/problems/RandomBlast/blast.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ constexpr double parsec_in_cm = C::parsec; // cm == 1 pc
 constexpr double m_H = C::m_p + C::m_e;	   // mass of hydrogen atom
 
 template <> struct Physics_Traits<RandomBlast> {
-	static constexpr bool is_self_gravity_enabled = false;
+	static constexpr bool is_self_gravity_enabled = true;
```
Comment (BenWibking) [inline src/problems/RandomBlast/blast.cpp:43]: Is self-gravity necessary for this test? Is it currently reproducible with hydro+AMR only?

Comment (chongchonghe) [reply src/problems/RandomBlast/blast.cpp:43]: Yes, gravity is definitely necessary for test of reproducibility of hydro+gravity+AMR. I can add another test for hydro+AMR only.

Comment (BenWibking) [reply src/problems/RandomBlast/blast.cpp:43]: A hydro+AMR only test would be useful to have.

Comment (chongchonghe) [reply src/problems/RandomBlast/blast.cpp:43]: Doen. The benchmark file is running

File: inputs/blast_unigrid_128_AMR_regression.in [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# Resolution and refinement
+# *****************************************************************
+amr.n_cell          = 128 128 128
+amr.max_level       = 1     # number of levels = max_level + 1
+amr.max_grid_size   = 128   # at least 128 for GPUs
+amr.blocking_factor = 64   # grid size must be divisible by this
+amr.n_error_buf     = 3     # minimum 3 cell buffer around tagged cells
+amr.grid_eff        = 0.7   # default
+
+do_reflux = 0
```
Comment (BenWibking) [inline inputs/blast_unigrid_128_AMR_regression.in]: Reflux needs to be on. Otherwise, conservation will fail.

Comment (chongchonghe) [reply inputs/blast_unigrid_128_AMR_regression.in]: OK, changed

**Review by BenWibking — APPROVED** [review]
